# HG changeset patch
# User Torsten <ttl@justmail.de>
# Date 1419682099 -3600
#      Sat Dec 27 13:08:19 2014 +0100
# Branch gui-release
# Node ID bd27cd403d005199dd43607fca1e1e3b7525bf02
# Parent  472a5572849c3d855085016279b379eefe5efaa8
fix errors when building octave for windows (regression from d93293218966)

* octave-dock-widget.cc (make_wodget): parameter dock readded since needed
  for windows;
  (set_style): declare icon_color for all cases, not only for custom styles,
  added missing plus sign

diff --git a/libgui/src/octave-dock-widget.cc b/libgui/src/octave-dock-widget.cc
--- a/libgui/src/octave-dock-widget.cc
+++ b/libgui/src/octave-dock-widget.cc
@@ -199,17 +199,17 @@ octave_dock_widget::make_window ()
 
 #endif
 
   _floating = true;
 }
 
 // dock the widget
 void
-octave_dock_widget::make_widget (bool)
+octave_dock_widget::make_widget (bool dock)
 {
 #if defined (Q_OS_WIN32)
 
   // windows: Since floating widget has no parent, we have to readd it
 
   QSettings *settings = resource_manager::get_settings ();
 
   // save last floating geometry if widget really was floating
@@ -283,26 +283,27 @@ octave_dock_widget::focusWidget ()
 
 void
 octave_dock_widget::set_style (bool active)
 {
   QString css;
   QString css_button;
   QString dock_icon;
 
+  QString icon_col = _icon_color;
+
   if (_floating)
     dock_icon = "widget-dock";
   else
     dock_icon = "widget-undock";
 
   if (_custom_style)
     {
 
       QColor bg_col, fg_col;
-      QString icon_col;
 
       if (active)
         {
           bg_col = _bg_color_active;
           fg_col = _fg_color_active;
           icon_col = _icon_color_active;
         }
       else
@@ -351,17 +352,17 @@ octave_dock_widget::set_style (bool acti
     }
 
 #if defined (Q_OS_WIN32)
   _title_widget->setStyleSheet (css);
   css_button = QString ("background: transparent; border: 0px;");
   _dock_button->setStyleSheet (css_button);
   _close_button->setStyleSheet (css_button);
   _dock_action->setIcon (QIcon (":/actions/icons/" + dock_icon + icon_col + ".png"));
-  _close_action->setIcon (QIcon (":/actions/icons/widget-close" + dock_icon + icon_col ".png"));
+  _close_action->setIcon (QIcon (":/actions/icons/widget-close" + dock_icon + icon_col + ".png"));
 #else
   setStyleSheet (css);
 #endif
 }
 
 void
 octave_dock_widget::handle_settings (const QSettings *settings)
 {
